test-mfs-lock.bash

```diff
@@ -0,0 +1,13 @@
+set -x
+set -v
+
+# The daemon needs to be running, otherwise the commands will compete on their
+# lock of the entire repo (not the MFS root) and fail.
+ipfs swarm addrs > /dev/null || (echo "daemon not running" && exit 1)
+
+ipfs --version
+ipfs files mkdir /test-lock/
+ipfs files rm /test-lock/ -r
+((echo "content" | ./cmd/ipfs/ipfs files write --create --parents --truncate --lock-time 3 /test-lock/file) && echo "ipfs write lock released" &)
```

Comment:
I think the way you've implemented the timeout the command `ipfs files write` does not exit until the timeout expires, consequently the lock is never present for the next command `ipfs repo gc`

Reply:
> the command ipfs files write does not exit until the timeout expires,

Correct. My bash is very poor but the idea was to have the above command running in the background while `ipfs repo gc` attempts to take the lock and has to wait until `ipfs write` releases it, where then I see both command's outputs roughly simultaneously.

But anyway, this is just a temporary ad-hoc test while in the draft stage; I still don't know how to actually test this in production to get this code ready to merge.